x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/parser/ExpressionTests.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.parser;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.Literal;
+import org.elasticsearch.xpack.sql.type.DataType;
+
+public class ExpressionTests extends ESTestCase {
+
+    private final SqlParser parser = new SqlParser();
+
+    public void testLiteralLong() throws Exception {
+        Expression lt = parser.createExpression(String.valueOf(Long.MAX_VALUE));
```

Comment:
Does it make sense to test for things like `+123` or `-123` (having the +/- sign added to the String expression)?

Reply:
Added a test for it (which uncovered an interesting case) - see below.